pkg/codec/renamer.go

```diff
@@ -0,0 +1,76 @@
+package codec
+
+import (
+	"reflect"
+	"unicode"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+var pkgPath = reflect.TypeOf(renamer{}).PkgPath()
+
+func NewRenamer(fields map[string]string) Modifier {
+	m := &renamer{
+		modifierBase: modifierBase[string]{
+			fields:           fields,
+			onToOffChainType: map[reflect.Type]reflect.Type{},
+			offToOnChainType: map[reflect.Type]reflect.Type{},
+		},
+	}
+	m.modifyFieldForInput = func(field *reflect.StructField, _, newName string) error {
+		field.Name = newName
+		if unicode.IsLower(rune(field.Name[0])) {
+			field.PkgPath = pkgPath
```

Comment:
What exactly happens here? Field that is unexported gets its PkgPath set to code pckg path. Why would an unexported field need PkgPath and why would we set it to codec pckg path?